docs/source/data-model.rst

```diff
@@ -1,18 +1,39 @@
-**********
-Data Model
-**********
+********************************
+Data Model: Documents and Events
+********************************
+
+*begin remove*
```

Comment:
I still like this original first paragraph, doesn't seem exactly redundant to what follows. Maybe combine it with the next paragraph?

Reply:
I spent some time trying to combine the original first paragraph and the text from documents.rst. My problem is that the original is talking about little-e "events" and it seems confusing to talk about big-e "Events" at the same time.

I will try again.

Reply:
Here is a new proposal: I added the original paragraph to the second new paragraph. I still think using the term "event-based" is confusing so I propose using "document-based" instead. I felt like the original text fits here since the following paragraph talks about subscribing to a stream of documents.

```
A primary design goal of bluesky is to enable better research by recording
rich metadata alongside measured data for use in later analysis. Documents are
how we do this. 

A *document* is our term for a Python dictionary with a schema created
by the RunEngine during plan execution.  All of the metadata and data generated
by executing the plan is organized into documents. Bluesky's document-based
data model supports complex, asynchronous data collection and enables
sophisticated live, prompt, streaming, and *post-facto* data analysis.

A :doc:"later section <callbacks>" describes how outside functions can
"subscribe" to a stream of these documents, visualizing, processing, or saving
them. This section provides an outline of documents themselves, aiming to give
a sense of the structure and familiarity with useful components.

```

Reply:
working on the third paragraph below....

Reply:
Hmm, I agree this is tricky. This line once made sense

> created by the RunEngine during plan execution

but is now too limiting: we can "ingest" data files from non-bluesky beamlines or simulations to create documents now. I wouldn't want to give the wrong impression that the RunEngine is the only way to mint documents.